tests/test_macro.py

```diff
@@ -1,12 +1,238 @@
+import pytest
+
 import numpy as np
+import pandas as pd
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
 
 from message_ix import Scenario, macro
+from message_ix.testing import make_westeros
+
+# tons of deprecation warnings come from reading excel (xlrd library), ignore
+# them for now
+pytestmark = pytest.mark.filterwarnings("ignore")
+
+MSG_ARGS = ('canning problem (MESSAGE scheme)', 'standard')
+
+W_DATA_PATH = Path(__file__).parent / 'data' / 'westeros_macro_input.xlsx'
+MR_DATA_PATH = Path(__file__).parent / 'data' / 'multiregion_macro_input.xlsx'
+
+
+class MockScenario:
+
+    def __init__(self):
+        self.data = pd.read_excel(MR_DATA_PATH, sheet_name=None)
+        for name, df in self.data.items():
+            if 'year' in df:
+                df = df[df.year >= 2030]
+                self.data[name] = df
+
+    def has_solution(self):
+        return True
+
+    def var(self, name, **kwargs):
+        df = self.data['aeei']
+        # add extra commodity to be removed
+        extra_commod = df[df.sector == 'i_therm']
+        extra_commod['sector'] = self.data['config']['ignore_sectors'][0]
```

Comment:
A little thing I noticed that doesn't change much but helps with the million Pandas Slice warnings.

> C:\Program Files\JetBrains\PyCharm 2019.1.1\helpers\pydev\pydevconsole.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

From my experience using assign insted of trying to overwrite a slice works very well. e.g. changing statements such as: 
`extra_commod['sector'] = self.data['config']['ignore_sectors'][0]` to 
`extra_commod = extra_commod.assign(sector=self.data['config']['ignore_sectors'][0])`